vscode-extension/src/extension.ts

```diff
@@ -42,6 +123,242 @@ export function deactivate() {
+  let closestDirectory = null;
+  const activeEditor = aiconfigEditorManager.getActiveEditor();
+  if (activeEditor?.document && !activeEditor.document.isUntitled) {
+    closestDirectory = path.dirname(activeEditor.document.fileName);
+  } else if (vscode.window.activeTextEditor?.document) {
+    closestDirectory = path.dirname(
+      vscode.window.activeTextEditor.document.fileName
+    );
+  } else {
+    closestDirectory = getCurrentWorkingDirectory(null);
+  }
+
+  let defaultModelRegistryPath = path.join(
+    closestDirectory,
+    "aiconfig_model_registry.py"
+  );
+
+  const modelRegistryPath = await vscode.window.showInputBox({
+    prompt: "Enter the path to create the model registry file",
+    value: defaultModelRegistryPath,
+    validateInput: (text) => {
+      if (!text) {
+        return "File path is required";
+      }
+
+      return null;
+    },
+  });
+
+  if (!modelRegistryPath) {
+    vscode.window.showInformationMessage("Model registry creation cancelled");
+    return;
+  }
+
+  // Create the model registry file from the sample
+  const sampleModelRegistryPath = vscode.Uri.joinPath(
+    context.extensionUri,
+    "static",
+    "example_aiconfig_model_registry.py"
+  );
+
+  try {
+    await vscode.workspace.fs.copy(
+      sampleModelRegistryPath,
+      vscode.Uri.file(modelRegistryPath),
+      { overwrite: false }
+    );
+  } catch (err) {
+    vscode.window.showErrorMessage(
+      `Error creating new file ${modelRegistryPath}. Error is ${err}`
+    );
+  }
+
+  const doc = await vscode.workspace.openTextDocument(modelRegistryPath);
+  if (doc) {
+    vscode.window.showTextDocument(doc);
+    vscode.window.showInformationMessage(
+      "Please customize your new model registry."
+    );
+  }
+
+  let config = vscode.workspace.getConfiguration(EXTENSION_NAME);
+  await handleCustomModelRegistryUpdate(
+    config,
+    aiconfigEditorManager,
+    modelRegistryPath
+  );
+}
+
+/**
+ * Registers (and persists) the custom model registry path with the extension and all open aiconfig editors.
+ */
+async function registerCustomModelRegistry(
+  aiconfigEditorManager: AIConfigEditorManager
+) {
+  let config = vscode.workspace.getConfiguration(EXTENSION_NAME);
+  let savedModelRegistryPath = config.get<string>("modelRegistryPath");
+
+  const modelRegistryPath = await vscode.window.showInputBox({
+    prompt: "Enter the path to the file containing your custom model registry",
+    value: savedModelRegistryPath,
+    validateInput: (text) => {
+      if (!text) {
+        return "File path is required";
+      }
+
+      if (!isValidFilePath(text)) {
+        return "File path is not valid or file does not exist";
+      }
+
+      return null;
+    },
+  });
+}
+
+async function handleCustomModelRegistryUpdate(
+  config: vscode.WorkspaceConfiguration,
+  aiconfigEditorManager: AIConfigEditorManager,
+  modelRegistryPath: string
+) {
+  // TODO: saqadri - ask the user if they want us to apply the setting globally
```

Comment:
+1, this will be important since I'd expect my configs to use the registered parsers in the same directory as them instead of using some global one that is registered once for use in a config